packages/@internationalized/date/src/string.ts

```diff
@@ -195,3 +199,70 @@ function offsetToString(offset: number) {
 export function zonedDateTimeToString(date: ZonedDateTime): string {
   return `${dateTimeToString(date)}${offsetToString(date.offset)}[${date.timeZone}]`;
 }
+
+/**
+ * Parses an ISO 8601 duration string (e.g. "P3Y6M6W4DT12H30M5S").
+ * @param value An ISO 8601 duration string.
+ * @returns A DateTimeDuration object.
+ */
+export function parseDuration(value: string): Required<DateTimeDuration> {
+  const match = value.match(DATE_TIME_DURATION_RE);
+
+  if (!match) {
+    throw new Error(`Invalid ISO 8601 Duration string: ${value}`);
+  }
+
+  const parseDurationGroup = (
+    group: string | undefined,
+    isNegative: boolean,
```

Comment:
Don't we have to support `-`, `+`, and no sign? I saw this on the Wikipedia page for ISO 8601.

Reply:
Yes, the [RegEx](https://github.com/adobe/react-spectrum/pull/3766/files#diff-aad5cb8398bdee14cf5a6bdc37c3665aabaa9ab6105fb0291003d6cbbade8031R25) allows for a `-` at the beginning of the duration string (as well as no sign), but doesn't allow for a `+` currently. According to the [TC39 temporal specification](https://tc39.es/proposal-temporal/#prod-ASCIISign), a `+` sign at the beginning of the duration string should be valid, so I will update the RegEx to allow for that sign.

Reply:
RegEx updated and additional tests added to validate `+` at the front of the duration string.